src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -104,6 +104,19 @@ public PublicClientApplicationBuilder WithDefaultRedirectUri()
             return this;
         }
 
+        /// <summary>
+        /// Configures the public client application to use multi cloud support.
+        /// It enables applications to pass in a default public cloud authority to the library and can still get tokens for resources from national clouds.
+        /// </summary>
+        /// <returns>A <see cref="PublicClientApplicationBuilder"/> from which to set more
+        /// parameters, and to create a public client application instance</returns>
+        public PublicClientApplicationBuilder WithMultiCloudSupport()
+        {
+            Config.MultiCloudSupport = true;
+            Config.ExtraQueryParameters.Add("instance_aware", "true");

```

Comment:
We should not add this kind of logic in the config layer. App developers may override this via `WithExtraQueryParameters(Dictionary)`

Reply:
The request can do this for example.